core/modules/url/store/actions.ts

```diff
@@ -7,23 +7,24 @@ import queryString from 'query-string'
 import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
 import { processMultipleDynamicRoutes, normalizeUrlPath, parametrizeRouteData } from '../helpers'
 import { storeCodeFromRoute, removeStoreCodeFromRoute } from '@vue-storefront/core/lib/multistore'
+import config from 'config'
 
 // it's a good practice for all actions to return Promises with effect of their execution
 export const actions: ActionTree<UrlState, any> = {
   // if you want to use cache in your module you can load cached data like this
   async registerMapping ({ commit }, { url, routeData }: { url: string, routeData: any}) {
     commit(types.REGISTER_MAPPING, { url, routeData })
-    await cacheStorage.setItem(url, routeData)
+    await cacheStorage.setItem(normalizeUrlPath(url), routeData)
     return routeData
   },
   /**
    * Register dynamic vue-router routes
    */
   async registerDynamicRoutes ({ state, dispatch }) {
     if (state.dispatcherMap) {
-      processMultipleDynamicRoutes(state.dispatcherMap)
+      processMultipleDynamicRoutes(state.dispatcherMap) // check if we're to add routes to vue router
       for (const [url, routeData] of Object.entries(state.dispatcherMap)) {
-        dispatch('registerMapping', { url, routeData })
+        await dispatch('registerMapping', { url, routeData })
```

Comment:
maybe `Promise.all` here? Could be faster with larger dispatcherMap object

Reply:
OK